examples/plugin/com_osvr_example_Tracker.cpp

```diff
@@ -0,0 +1,122 @@
+/** @date 2015
+    @author
+    Sensics, Inc.
+    <http://sensics.com/osvr>
+*/
+
+// Copyright 2015 Sensics Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Internal Includes
+#include <osvr/PluginKit/PluginKit.h>
+#include <osvr/PluginKit/TrackerInterfaceC.h>
+
+// Generated JSON header file
+#include "com_osvr_example_Tracker_json.h"
+
+// Library/third-party includes
+
+// Standard includes
+#include <iostream>
+#include <chrono>
+#include <thread>
+#include <random>
+#include <ctime>
+
+// Anonymous namespace to avoid symbol collision
+namespace {
+
+class TrackerDevice {
+  public:
+    TrackerDevice(OSVR_PluginRegContext ctx) {
+        /// Create the initialization options
+        OSVR_DeviceInitOptions opts = osvrDeviceCreateInitOptions(ctx);
+
+        // configure our tracker
+        osvrDeviceTrackerConfigure(opts, &m_tracker);
+
+        /// Create the device token with the options
+        m_dev.initAsync(ctx, "Tracker", opts);
+
+        /// Send JSON descriptor
+        m_dev.sendJsonDescriptor(com_osvr_example_Tracker_json);
+
+        /// Register update callback
+        m_dev.registerUpdateCallback(this);
+
+        /// Seed pseudo random generator.
+        std::srand(std::time(0));
+    }
+
+    OSVR_ReturnCode update() {
+
+        std::this_thread::sleep_for(std::chrono::milliseconds(
+            5)); // Simulate waiting a quarter second for data.
+
+        OSVR_TimeValue now;
+        osvrTimeValueGetNow(&now);
+        double t = 5.0 * ((double)now.seconds + ((double)now.microseconds / 1000000.0));
```

Comment:
I don't understand the purpose of this line: it looks like it's trying to get the time since epoch in seconds, but then multiply it by 5.  If you want time since startup, you can either use `std::chrono` stuff (some of this would go in the class, some in the constructor, some in the update) or `OSVR_TimeValue`, but I'd recommend `chrono`

```c++
using our_clock = std::chrono::steady_clock; // like a typedef, stick it somewhere useful.

our_clock::time_point m_startTime; // class member
m_startTime(our_clock::now()) // in the constructor initializers

// then in update,
std::chrono::duration<double> secondsSinceStart = our_clock::now() - m_startTime;
double t = secondsSinceStart.count(); // doesn't do the * 5
```

then if you want a function of `sin(5 * t)`, do it like that, because `t` sounds like time, not time with a scale factor. (and yes, I understand that this probably originated in a similar simulated/dummy data plugin of another kind - I just didn't get asked for a review on it and/or didn't do a good enough one on it)